assignment-client/src/messages/MessagesMixer.cpp

```diff
@@ -62,13 +69,19 @@ void MessagesMixer::handleMessages(QSharedPointer<ReceivedMessage> receivedMessa
 void MessagesMixer::handleMessagesSubscribe(QSharedPointer<ReceivedMessage> message, SharedNodePointer senderNode) {
     QString channel = QString::fromUtf8(message->getMessage());
     _channelSubscribers[channel] << senderNode->getUUID();
+
+    _allSubscribers[senderNode->getUUID()] << 0;
 }
 
 void MessagesMixer::handleMessagesUnsubscribe(QSharedPointer<ReceivedMessage> message, SharedNodePointer senderNode) {
     QString channel = QString::fromUtf8(message->getMessage());
     if (_channelSubscribers.contains(channel)) {
         _channelSubscribers[channel].remove(senderNode->getUUID());
     }
+
+    if (_allSubscribers.contains(senderNode->getUUID())) {
+        _allSubscribers.remove(senderNode->getUUID());
+    }
```

Comment:
What say a node has subscribed to two channels, then unsubscribes from one. This code will remove the subscriber totally from `_allSubscribers`.